### PR TITLE
auth-challenge

### PR DESCRIPTION
**Pull request by elvisonob**

### CODE REVIEW

File: src/client/App.jsx [hunk 1]
```diff
@@ -15,16 +15,68 @@ function App() {
...
+        password: password
+      })
+    }
+
+    fetch ('http://localhost:4000/user/register', options)
+    .then(res => res.json())
+    .then(json => {
+      console.log('handleRegister', json)
+
+    })
```
Comment (BrandonV21) [inline src/client/App.jsx:34]: Well structured. Nice job in making a fetch request.

File: src/server/controllers/user.js [hunk 2]
```diff
@@ -8,27 +8,41 @@ const jwtSecret = 'mysecret';
 const register = async (req, res) => {
     const { username, password } = req.body;
 
-    const createdUser = null;
+    const hashedPassword = await bcrypt.hash(password, 10)
```
Comment (BrandonV21) [inline src/server/controllers/user.js:11]: It's great that you hashed the password before the user is created!

File: src/client/App.jsx [hunk 3]
```diff
@@ -15,16 +15,68 @@ function App() {
...
+        description: description,
+        runtimeMins: runtimeMins 
+      })
+    }
+  
+
+    fetch ('http://localhost:4000/user/movie', options)
+    .then(res => res.json())
+    .then(json => {
+      console.log('handlecreatemovie', json)
```
Comment (Khateeb1111) [inline src/client/App.jsx:75]: I like that you console logged your function. Good stuff :)

File: src/server/controllers/movie.js [hunk 4]
```diff
@@ -14,13 +14,19 @@ const createMovie = async (req, res) => {
     const { title, description, runtimeMins } = req.body;
 
     try {
-        const token = null;
```
Comment (New-Arday) [inline src/server/controllers/movie.js:17]: You still have your token set as null - how did you manage to get a working results, I wonder

File: src/server/controllers/user.js [hunk 5]
```diff
@@ -8,27 +8,41 @@ const jwtSecret = 'mysecret';
...
     }
 
-    const passwordsMatch = false;
+    const passwordsMatch = await bcrypt.compare (password, foundUser.password)
 
     if (!passwordsMatch) {
         return res.status(401).json({ error: 'Invalid username or password.' });
     }
 
-    const token = null;
```
Comment (New-Arday) [inline src/server/controllers/user.js:31]: Same thing here, I saw this with someone else's code as well so I think I'm missing the understanding of this code. I had to modify this code to work where you have left it as null here.

**Pull request closed without merging**